### PR TITLE
Add bucket policy tests

### PR DESCRIPTION
**Pull request by sagihirshfeld**
This PR adds basic bucket policy tests and the framework components for implementing them.

Passing verification run: https://url.corp.redhat.com/d0723f7

### CODE REVIEW

**Review by vavuthu — APPROVED** [review]

File: framework/bucket_policies/s3_operation_access_tester.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from framework.bucket_policies.access_validation_strategy_factory import (
+    AccessValidationStrategyFactory,
+)
+from noobaa_sa.exceptions import UnexpectedBehaviour
+
+
+class S3OperationAccessTester:
```
Comment (udaysk23) [inline framework/bucket_policies/s3_operation_access_tester.py:7]: @sagihirshfeld Can't we skip writing new class and moving `check_client_access_to_bucket_op` method to bucket_utils and calling it from there

**Review by udaysk23 — APPROVED** [review]
Very nicely written code @sagihirshfeld :)

File: framework/bucket_policies/access_validation_strategy_factory.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        Returns:
+            AccessValidationStrategy: An instance of the appropriate strategy
+
+        Raises:
+            NotImplementedError: If the operation is not supported
+        """
+
+        # Dynamically import the strategy module
+        try:
+            prefix = "framework.bucket_policies.access_validation_strategies."
```
Comment (ebenahar) [inline framework/bucket_policies/access_validation_strategy_factory.py:40]: as discussed in person, please change to a dynamic way of setting the prefix.
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/factory.py should be useful for reference

Not a PR merge blocker, so you can do that in a follow up PR

**Review by ebenahar — APPROVED** [review]

File: framework/bucket_policies/access_validation_strategies/copy_object_validation_strategy.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    """
+    A strategy for validating access to the CopyObject operation
+    """
+
+    @property
+    def expected_success_code(self):
+        return 200
+
+    def setup(self, **setup_kwargs):
+        self.test_obj_key = generate_unique_resource_name(prefix="test-obj-")
```
[the remark this answers is not in the document]
Comment (sagihirshfeld) [reply framework/bucket_policies/access_validation_strategies/copy_object_validation_strategy.py:17]: Addressed in https://github.com/red-hat-storage/noobaa-sa-ci/pull/33, specifically in commit https://github.com/red-hat-storage/noobaa-sa-ci/pull/33/commits/d7e424cd0269c94979c73870ae5ccb05ee44d97d

File: framework/bucket_policies/access_validation_strategy_factory.py (hunk 2 above)
Comment (sagihirshfeld) [reply framework/bucket_policies/access_validation_strategy_factory.py:40]: Addressed in https://github.com/red-hat-storage/noobaa-sa-ci/pull/33, specifically in commit https://github.com/red-hat-storage/noobaa-sa-ci/pull/33/commits/dacf4a0fd50f8dbf7d34dc26e525778a32cb2a3b

**Pull request merged**